csrc/scheduler/matmul.cpp

```diff
@@ -700,18 +700,26 @@ void scheduleMatmul(Fusion* fusion, const MatmulParams& params) {
   auto cached_and_forked_outputs =
       scheduler_utils::cacheAndForkOutputs(fusion, should_unroll);
 
-  const auto& roles_map_opt = mma_utils::getTensorsRoles(fusion);
-
-  // NOTE: the contents of roles_map have been already validated during
-  //  compute-time checks
-  NVF_ERROR(roles_map_opt.isValid(), roles_map_opt.getErrorMsg());
-  const auto roles_map = roles_map_opt.getData();
+  mma_utils::CombineMulSum combiner(fusion);
+  std::vector<mma_utils::MulSumProperties> mulSum =
+      combiner.generateMulSumCanidates();
 
   auto mma_ops = ir_utils::getOpsOfType<MmaOp>(fusion);
+  if (combiner.isValid() && mma_ops.empty()) {
+    combiner.replaceWithMmaOp();
+  }
+
+  mma_ops = ir_utils::getOpsOfType<MmaOp>(fusion);
```

Comment:
With the proposed changes in public API of `CombineMulSum` there won't be a need for `mul_sum` object.

```suggestion
  mma_utils::CombineMulSum combiner(fusion);

  auto mma_ops = ir_utils::getOpsOfType<MmaOp>(fusion);
  if (mma_ops.empty()) {
    // NOTE: the combiner object has been already validated during
    //  previous checks performed on the current fusion
    combiner.replaceWithMmaOp();
  }

  mma_ops = ir_utils::getOpsOfType<MmaOp>(fusion);
```

Reply:
Thanks!
I made most of those changes. 
In addition, I used:

```
  auto mma_ops = ir_utils::getOpsOfType<MmaOp>(fusion);
  if (combiner.isValid() && mma_ops.empty()) {
    combiner.replaceWithMmaOp();
    mma_ops = ir_utils::getOpsOfType<MmaOp>(fusion);
  }
```

I kept the `isValid` check to be safe, since I think function can be used directly.
I also moved the (second) getOps to inside the if-condidion.